File: content/1-basics/5-while-loop/cpp/while_loop.cpp
#include <iostream>

int main()
{
    int number;

    std::cout << "Enter numbers (enter 0 to stop):" << std::endl;

    while (true)
    {
        std::cout << "Enter a number: ";
        std::cin >> number;

        if (number == 0)
        {
            break;
        }

        std::cout << "You entered: " << number << std::endl;
    }

    std::cout << "Loop terminated because you entered 0." << std::endl;

    return 0;
}

File: content/1-basics/5-while-loop/go/while_loop.go
package main

import (
    "fmt"
)

func main() {
    var number int

    fmt.Println("Enter numbers (enter 0 to stop):")

    for {
        fmt.Print("Enter a number: ")
        fmt.Scan(&number)

        if number == 0 {
            break
        }

        fmt.Println("You entered:", number)
    }

    fmt.Println("Loop terminated because you entered 0.")
}